chain/block.go

```diff
@@ -823,12 +826,41 @@ func (b *StatelessBlock) View(ctx context.Context, blockRoot *ids.ID, verify boo
 		}
 		return b.view, nil
 	}
+
+	// Handle case where we need to process block.
 	b.vm.Logger().Info("block not processed",
 		zap.Uint64("height", b.Hght),
 		zap.Stringer("blkID", b.ID()),
-		zap.Bool("attemptVerify", blockRoot != nil),
+		zap.Stringer("blockRoot", blockRoot),
+		zap.Bool("verify", verify),
 	)
+	acceptedState, err := b.vm.State()
+	if err != nil {
+		return nil, err
+	}
+	acceptedRoot, err := acceptedState.GetMerkleRoot(ctx)
+	if err != nil {
+		return nil, err
+	}
 	if blockRoot == nil {
+		// If the block is not processed but [acceptedState]
+		// contains the height of the block, we should return the accepted
+		// state.
+		//
+		// This can happen when we are building/verifying a child block
+		// immediately after restart (will not have in-memory view to indicate
+		// it is processed).
+		acceptedHeightRaw, err := acceptedState.Get(HeightKey(b.vm.StateManager().HeightKey()))
+		if err != nil {
+			return nil, err
+		}
+		acceptedHeight := binary.BigEndian.Uint64(acceptedHeightRaw)
```

Comment:
Would it be safer to use blockID here?

Reply:
If the accepted height is in state, it seems very broken if this doesn't match the block we called `View` on...